### PR TITLE
#163052933 Implement login functionality

### PR DESCRIPTION
**Pull request by abejide001**
#### What does this PR do?
Returns JWT(json web token) to user on successful signin
#### Description of Task to be completed?
Handle errors for bad login details
Send jwt to authorized users
Fix jsdoc comments errors
Write test for the login functionality
#### How should this be manually tested?
1. Run `npm i` to install application dependencies
2. Create a .env file and copy the content in .env.sample to create your custom configuration
3. Create a databse for test and development by running `createdb dbname`
4. Run `sequelize db:seed:all` to seed a user into the database
5. Run `npm run start:dev` to start the application locally
6. Open postman and send a request to the endpoint `http://localhost:9000/api/v1/auth/signin` with the `email` and `password` as the key
7. Run `npm test`
#### Any background context you want to provide?
The JWT payload contains
- id
- username
- userEmail
- signUpType
#### What are the relevant pivotal tracker stories?
[#163052933](https://www.pivotaltracker.com/story/show/163052933)
#### Screenshots (if appropriate)
screen shot 2019-01-12 at 10 02 43 am

### CODE REVIEW

File: src/controllers/UsersController.js [hunk 1]
```diff
@@ -1,90 +1,159 @@
+/* eslint-disable valid-jsdoc */
+import bcrypt from 'bcryptjs';
```
Comment (andela-dbamidele) [inline src/controllers/UsersController.js]: Please place the password function in the helper class.

File: src/db/migrations/20190110043410-create-user.js [hunk 2]
```diff
@@ -1,61 +1,57 @@
...
+    createdAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
+    updatedAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    }
+  }),
+  down: (queryInterface, Sequelize) => queryInterface.dropTable('Users')
```
[the remark this answers is not in the document]
Comment (andela-dbamidele) [reply src/db/migrations/20190110043410-create-user.js]: Please take note of houndci errors.

File: src/db/migrations/20190110143725-create-profile.js [hunk 3]
```diff
@@ -1,51 +1,47 @@
...
+    createdAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
+    updatedAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    }
+  }),
+  down: (queryInterface, Sequelize) => queryInterface.dropTable('Profiles')
```
[the remark this answers is not in the document]
Comment (andela-dbamidele) [reply src/db/migrations/20190110143725-create-profile.js]: Same here.

File: src/helpers/TokenAuthenticate.js [hunk 4]
```diff
@@ -3,35 +3,50 @@ import dotenv from 'dotenv';
...
+    const token = req.headers.Authorization || req.headers['x-access-token'] || req.query.token || req.body.token;
+    if (!token) {
+      return res.status(401).send({
+        status: 'error',
+        message: 'No token provided',
+      });
     }
-
+    try {
+      const verifyUser = await jwt.verify(token, keyconfig);
```
[the remark this answers is not in the document]
Comment (andela-dbamidele) [reply src/helpers/TokenAuthenticate.js]: Please fix this if you can.

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]
Sorry I forgot to push the review button. Please make those changes.

File: src/helpers/TokenAuthenticate.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (abejide001) [reply src/helpers/TokenAuthenticate.js]: Fixed

File: src/db/migrations/20190110143725-create-profile.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (abejide001) [reply src/db/migrations/20190110143725-create-profile.js]: Fixed

File: src/db/migrations/20190110043410-create-user.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (abejide001) [reply src/db/migrations/20190110043410-create-user.js]: Fixed

File: src/controllers/UsersController.js (hunk 1 above)
Comment (abejide001) [reply src/controllers/UsersController.js]: I have moved the logic for the password compare into the helper file

**Review by andela-dbamidele — APPROVED** [review]

**Pull request merged**